File: app/src/main/java/by/toggi/rxbsuir/di/OkHttpClientModule.java
package by.toggi.rxbsuir.di;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;
import okhttp3.Cache;
import okhttp3.Interceptor;
import okhttp3.OkHttpClient;

@Module
@InstallIn(SingletonComponent.class)
public class OkHttpClientModule {

    @Provides
    @Singleton
    OkHttpClient provide(@Named("logging") Interceptor interceptor, Cache cache) {
        return new OkHttpClient.Builder().cache(cache).addInterceptor(interceptor).build();
    }
}

File: app/src/main/kotlin/by/toggi/rxbsuir/di/OkHttpClientModule.kt
package by.toggi.rxbsuir.di

import android.util.Log
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object OkHttpClientModule {

    @Provides
    @Singleton
    fun provide(): OkHttpClient {
        val interceptor = HttpLoggingInterceptor { Log.v("OkHttpClient", it) }
        interceptor.level = HttpLoggingInterceptor.Level.BASIC
        return OkHttpClient.Builder()
            .addInterceptor(interceptor)
            .build()
    }
}
